File: src/main/java/com/nugurang/graphql/mutation/TeamMutation.java
package com.nugurang.graphql.mutation;

import com.nugurang.dto.TeamDto;
import com.nugurang.dto.TeamInputDto;
import com.nugurang.service.TeamService;
import graphql.kickstart.tools.GraphQLMutationResolver;
import org.springframework.stereotype.Service;

@Service
public class TeamMutation implements GraphQLMutationResolver {
    private final TeamService teamService;

    public TeamDto createTeam(TeamInputDto teamInputDto) {
        return teamService.createTeam(teamInputDto).toDto();
    }

    public TeamDto updateTeam(TeamInputDto teamInputDto, Long teamId) {
        return teamService.updateTeam(teamInputDto, teamId).toDto();
    }

    public Long deleteTeam(Long teamId) {
        teamService.deleteTeam(teamId);
        return teamId;
    }

    //<editor-fold defaultstate="collapsed" desc="delombok">
    @SuppressWarnings("all")
    
    public TeamMutation(final TeamService teamService) {
        this.teamService = teamService;
    }
    //</editor-fold>
}

File: src/main/kotlin/com/nugurang/graphql/mutation/TeamMutation.kt
package com.nugurang.graphql.mutation

import com.nugurang.dto.TeamDto
import com.nugurang.dto.TeamInputDto
import com.nugurang.mapper.TeamMapper
import com.nugurang.service.TeamService
import graphql.kickstart.tools.GraphQLMutationResolver
import org.springframework.stereotype.Service

@Service
class TeamMutation(
    private val teamService: TeamService,
    private val teamMapper: TeamMapper
) : GraphQLMutationResolver {
    fun createTeam(teamInputDto: TeamInputDto): TeamDto {
        return teamService
            .createTeam(teamInputDto)
            .let(teamMapper::toDto)
    }

    fun updateTeam(teamInputDto: TeamInputDto, teamId: Long): TeamDto {
        return teamService
            .updateTeam(teamInputDto, teamId)
            .let(teamMapper::toDto)
    }

    fun deleteTeam(teamId: Long): Long {
        teamService.deleteTeam(teamId)
        return teamId
    }
}